orca-core/src/main/groovy/com/netflix/spinnaker/orca/batch/adapters/TaskTasklet.groovy

```diff
@@ -70,6 +70,7 @@ class TaskTasklet implements Tasklet {
     try {
       if (stage.execution.canceled) {
         setStopStatus(chunkContext, ExitStatus.STOPPED, ExecutionStatus.CANCELED)
+        contribution.exitStatus = ExitStatus.STOPPED
```

Comment:
It's an alternative to doing the `setTerminateOnly()` thing – indicates that the Job has stopped because of an interruption but doesn't cause the problems with early / never firing listeners.